app/models/course/lecture.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+class Course::Lecture < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  has_many_attachments on: :content
+
+  belongs_to :course, inverse_of: :lectures
+
+  scope :sorted_by_date, -> { order(start_at: :desc) }
+
+  TIME_BRAINCERT = '%I:%M%p'
+  DATE_ISO = '%Y-%m-%d'
+  BRAINCERT_API_BASE_URL = 'https://api.braincert.com'
+
+  def handle_access_link(user, is_instructor)
```

Comment:
I would think that `handle_access_link` should be placed in a service then called in the controller, together with:
 - `generate_classroom_link`
 - `generate_classroom_link_params`
 - `post`
 - `call_braincert_api`
 - `create_classroom`
 - `create_classroom_params`